Prototyping/Sit/sit.js

```diff
@@ -0,0 +1,692 @@
+                MyAvatar.restoreRoleAnimation(roles[i]);
+            }
+            MyAvatar.characterControllerEnabled = true;
+            MyAvatar.hmdLeanRecenterEnabled = true;
+
+            Script.setTimeout(function () {
+                MyAvatar.centerBody();
+            }, STANDUP_DELAY);
+
+        }
+        sittingDown = false;
+        deadlockGate = false;
+        sitDownSettlePeriod = null;
+    };
+
+    // User can also click on overlay to sit down
+    this.mouseReleaseOnOverlay = function (overlayID, pointerEvent) {
+        print("Mouse release");
+        if (overlayID === overlaySittable && pointerEvent.isLeftButton) {
+            print("CAN SIT", deadlockGate, Settings.getValue(SETTING_KEY));
+            deadlockGate = true;
+            if (Settings.getValue(SETTING_KEY) === "") {
+                sitDown();
+            } else {
+                deadlockGate = false;
+            }
+        }
+    };
+
+    function update(dt) {
+
+        var hasAvatarSpineError = false;
+        var hasHeldDriveKey = false;
+        var hasAvatarMovedTooFar = false;
+
+        if (sittingDown === true && canStand) {
+
+            if (overlaySittable !== null) {
+                overlays.sittable.remove();
+            }
+
+            var now = Date.now();
+
+            // ACTIVE DRIVE KEY
+            // Check if a drive key is pressed
+            var hasActiveDriveKey = false;
+
+            for (var i in OVERRIDDEN_DRIVE_KEYS) {
+                if (MyAvatar.getRawDriveKey(OVERRIDDEN_DRIVE_KEYS[i]) != 0.0) {
+                    hasActiveDriveKey = true;
+
+                    if (driveKeyPressedStart === null) {
+                        driveKeyPressedStart = now;
+                    }
+
+                    if (overlayStandUp === null) {
+                        overlays.standUp.create();
+                    }
+
+                    break;
+                }
+            }
+
+            // Only standup if user has been pushing a drive key for RELEASE_TIME
+            if (hasActiveDriveKey) {
+
+                if (driveKeyPressedStart !== null) {
+                    overlays.standUp.update();
+                    var elapsed = now - driveKeyPressedStart;
+                    hasHeldDriveKey = elapsed > RELEASE_TIME;
+                }
+
+            } else {
+                if (overlayStandUp) {
+                    driveKeyPressedStart = null;
+                    overlays.standUp.remove();
+                }
+            }
+
+            // AVATAR DISTANCE 
+            var avatarDistance = Vec3.distance(MyAvatar.position, seatCenterPosition);
+            if (avatarDistance > DISTANCE_FROM_CHAIR_TO_STAND_AT && sitDownSettlePeriod && now > sitDownSettlePeriod) {
+                hasAvatarMovedTooFar = true;
+            }
+
+            // AVATAR SPINE ERROR
+            if (HMD.active) {
+
+                var headWorldPosition;
+                if (MyAvatar.getJointIndex("Head") !== -1) {
```

Comment:
I don't think we need to check if the avatar has a head here.  If we are in HMD mode we are guaranteed that getPoseValue(Head) will return something.